include/buzzer.h

```diff
@@ -0,0 +1,50 @@
+#include <stdint.h>
+#include "hal_general.h"
```

Comment:
You don't need to include hal_general.h here and you certainly don't need it twice